### DISCUSSION TITLE
Any so far known gotchas with making a flash 8 game and distributing with the desktop application?

### DISCUSSION THREAD

**Discussion by ghost** [ruffle-rs/ruffle#18637] [General]
I suppose this question can also apply to AS3 as well, but I'm specifying flash 8 AS2 because that's the one I like to use. Any gotchas right now about using flash 8 (Or CS6 if talking about AS3) to distribute on steam? Would saving offline without any kind of external connection or dependency on external stuff be possible as well? I know it most likely is, just wanted to hear information from the devs about how things are going so far.

**Comment by Croworbit** [comment]
I think everything is possible, but no one has done a ruffle flash game on steam yet

ruffle can do save data if your game creates it, and offline would probably be easier than anything online

I don't think there would be much dependencies as long as your game isn't fragmented or hosted in the cloud, I think in theory you'd just need the ruffle binary and your game, but probably also something like a batch file to load the ruffle settings your game requires

**Reply by ghost** [reply]
I've done considerable digging into the older flash player docs. There existed at one point (Maybe still does?) a C extension API you compile with a C99 valid compiler. Will that work with ruffle? I probably won't need it unless I'm doing seriously heavy number crunching, but you never know when you will.

**Reply by Croworbit** [reply]
is that like the alchemy/crossbridge? 
I think most games using that don't perform well at the moment with ruffle

**Reply by ghost** [reply]
No, I don't think it is at least, the C API was really just a short chapter in their docs. The API was just a single C header file. They did publish an example project on their website, thankfully was still downloadable from archive.org. I'll post it right here for you to see what I mean.
[clevel_ext_sample.zip](https://github.com/user-attachments/files/17793506/clevel_ext_sample.zip)

**Reply by Croworbit** [reply]
I'm not sure if many things using that have been tested, so might be uncharted waters, can you link the old docs here, so the developers might better answer this?

**Reply by ghost** [reply]
Sure! Most conveniently, macromedia made a downloadable zip of all their docs in PDF format. You can see the last modified date was ranged from september-october 2005. I don't know if adobe ever made use of the C API by doing something such as implementing a newer C++ API, but that's a question for another day I suppose. If you open fl8_extending.pdf, you'll see a short chapter all about the C-Extension API detailing about each macro, typedef, function, and so on and so forth. This may be very useful to anyone who wants to turbo charge flash applications in C or maybe possibly even C++.

I'll give you the direct link because it's too big to upload the zipped up PDF files.
https://web.archive.org/web/20051124203912/http://download.macromedia.com/pub/documentation/en/flash/fl8/flash_pdfs.zip
